tiledb/sm/query/dimension_label/index_data.h

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file index_data.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Class for internally managed index data for dimension labels.
```

Comment:
This seems like fairly generic container that should be moved upward. It could also potentially be simplified, I think -- possibly even as a type alias around `std::vector` plus a few factory methods for that alias (looks like Range -> forward to std::iota; Datatype + num values).

Reply:
Should I move this to `tiledb/type`? Are there any other specific changes you would like me to make at this time?

Reply:
On second thought, if we're not going to refactor it, then I'd suggest to leave it here for now. We already have many of these sort of one-off classes, and I'd like to keep that number from growing. The problem I see is that this looks like a light wrapper around a `std::vector`, but it lacks the generality of a container (eg iterator interface) while being closely tied to `Range`. I'd like to understand whether we really need this, or if a type alias to vector plus the factory functions would be sufficient for now.

Reply:
Marking resolved after offline discussions -- current code works, we'll keep it isolated in dimensionlabel, and generalize as we find similar needs in the future. Thanks.